backend/infrahub/graphql/queries/task.py

```diff
@@ -24,30 +25,53 @@ async def resolve(
         info: GraphQLResolveInfo,
         limit: int = 10,
         offset: int = 0,
-        ids: Optional[list] = None,
-        related_node__ids: Optional[list] = None,
+        ids: list | None = None,
+        branch: str | None = None,
+        related_node__ids: list | None = None,
     ) -> dict[str, Any]:
         related_nodes = related_node__ids or []
         ids = ids or []
-        return await Tasks.query(info=info, limit=limit, offset=offset, ids=ids, related_nodes=related_nodes)
+        return await Tasks.query(
+            info=info, branch=branch, limit=limit, offset=offset, ids=ids, related_nodes=related_nodes
+        )
 
     @classmethod
     async def query(
-        cls, info: GraphQLResolveInfo, limit: int, offset: int, related_nodes: list[str], ids: list[str]
+        cls,
+        info: GraphQLResolveInfo,
+        limit: int,
+        offset: int,
+        related_nodes: list[str],
+        ids: list[str],
+        branch: str | None,
     ) -> dict[str, Any]:
         context: GraphqlContext = info.context
         fields = await extract_fields_first_node(info)
 
-        return await TaskNode.query(
-            db=context.db, fields=fields, limit=limit, offset=offset, ids=ids, related_nodes=related_nodes
+        # During the migration, query both Prefect and Infrahub to get the list of tasks
+        if not branch:
+            infrahub_tasks = await TaskNode.query(
+                db=context.db, fields=fields, limit=limit, offset=offset, ids=ids, related_nodes=related_nodes
+            )
+        else:
+            infrahub_tasks = {}
+
+        prefect_tasks = await TaskNewNode.query(
+            fields=fields, branch=branch, related_nodes=related_nodes, limit=limit, offset=offset
         )
 
+        return {
+            "count": infrahub_tasks.get("count", 0) + prefect_tasks.get("count", 0),
+            "edges": infrahub_tasks.get("edges", []) + prefect_tasks.get("edges", []),
```

Comment:
How will this work? :) Will we return 20 entries if we have a limit of 10? Might be fine during the migration provided that the frontend doesn't get confused and the migration doesn't take that long to complete.

Reply:
Yes I forgot to mention this one, indeed limit is not gonna work completely during the migration but I don't think it will be too much of an issue